cartographer/common/task_test.cc

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cartographer/common/task.h"
+
+#include <memory>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace cartographer {
+namespace common {
+namespace {
+
+class MockThreadPool : public ThreadPoolInterface {
+ public:
+  virtual MOCK_METHOD1(NotifyReady, void(Task*));
+  MOCK_METHOD1(Schedule, void(const std::function<void()>&));
```

Comment:
`const Task::WorkItem&`?

Reply:
This is the old interface, it will be removed soon.